core/src/executor/evaluate/function.rs

```diff
@@ -605,6 +605,37 @@ pub fn generate_uuid<'a>() -> Evaluated<'a> {
     Evaluated::from(Value::Uuid(Uuid::new_v4().as_u128()))
 }
 
+pub fn greatest(_name: String, exprs: Vec<Evaluated<'_>>) -> Result<Evaluated<'_>> {
+    let exprs: Vec<Value> = exprs
+        .into_iter()
+        .filter_map(|value| match value.try_into().unwrap() {
+            Value::Null => None,
+            value => Some(value),
+        })
+        .collect();
+    if exprs.is_empty() {
+        return Ok(Evaluated::from(Value::Null));
+    }
+    let first_type = &exprs[0].get_type();
+
+    if !exprs.iter().all(|value| value.get_type() == *first_type) {
+        return Err(EvaluateError::CannotCompareDifferentTypes.into());
+    }
+
+    let greatest_value = exprs.into_iter().max_by(|a, b| match (a, b) {
+        (Value::I64(a), Value::I64(b)) => a.cmp(b),
+        (Value::F64(a), Value::F64(b)) => a.partial_cmp(b).unwrap_or(std::cmp::Ordering::Equal),
+        (Value::Str(a), Value::Str(b)) => a.cmp(b),
+        (Value::Date(a), Value::Date(b)) => a.cmp(b),
+        _ => panic!("Unsupported type in comparison"),
```

Comment:
We do not recommend using panic. Therefore, it would be better to implement a separate error object for unsupported types.